### ISSUE TITLE
Deconnection action obs.simulate has side effects on obs

### ISSUE THREAD

**Issue by mjothy** [rte-france/Grid2Op#113]
## Environment
 - Grid2op version: `0.9.3`
 - System: `windows`

## Bug description
A deconnection Action on an Observation seems to have side effects that make further Action on the same Observation output a DivergingPowerFlow exception

## How to reproduce
Just run the code snippet

### Code snippet
```python
import grid2op
env = grid2op.make("l2rpn_2019")
obs = env.get_obs()
#action test snipet
act1 = env.action_space({"set_bus": {'loads_id': [(3, 2)], 'lines_or_id': [(9, 2)], 'lines_ex_id': [(1, 1), (4, 1), (6, 1)]}})
act2 = env.action_space({"set_bus": {'loads_id': [(3, 1)], 'lines_or_id': [(9, 1)], 'lines_ex_id': [(1, 2), (4, 2), (6, 2)]}})
simul_obs, simul_r1, d1, info1 = obs.simulate(act1)
simul_ob2, simul_r2, d2, info2 = obs.simulate(act2)
print(info1)
print(info2)        
#deconnection action that breaks the env
deconexion_action = env.action_space({"set_line_status": [(9, -1)]})
obs_linecut, reward, done, info = obs.simulate(deconexion_action)
#action test snipet
act1 = env.action_space({"set_bus": {'loads_id': [(3, 2)], 'lines_or_id': [(9, 2)], 'lines_ex_id': [(1, 1), (4, 1), (6, 1)]}})
act2 = env.action_space({"set_bus": {'loads_id': [(3, 1)], 'lines_or_id': [(9, 1)], 'lines_ex_id': [(1, 2), (4, 2), (6, 2)]}})
simul_obs, simul_r1, d1, info1 = obs.simulate(act1)
simul_ob2, simul_r2, d2, info2 = obs.simulate(act2)
print(info1)
print(info2)        
print("done")
```
## Current output
```
{'disc_lines': array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False]), 'is_illegal': False, 'is_ambiguous': False, 'is_dispatching_illegal': False, 'is_illegal_reco': False, 'exception': [], 'rewards': {}}
{'disc_lines': array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False]), 'is_illegal': False, 'is_ambiguous': False, 'is_dispatching_illegal': False, 'is_illegal_reco': False, 'exception': [], 'rewards': {}}
{'disc_lines': None, 'is_illegal': False, 'is_ambiguous': False, 'is_dispatching_illegal': False, 'is_illegal_reco': False, 'exception': [Grid2OpException DivergingPowerFlow DivergingPowerFlow('Powerflow has diverged during computation.')], 'rewards': {}}
{'disc_lines': None, 'is_illegal': False, 'is_ambiguous': False, 'is_dispatching_illegal': False, 'is_illegal_reco': False, 'exception': [Grid2OpException DivergingPowerFlow DivergingPowerFlow('Powerflow has diverged during computation.')], 'rewards': {}}
done
```

**Comment by Tezirg** [comment]
I can confirm this is a reproducible bug.
Doing the following seems to solve the issue:
```python
        obs_linecut, reward, done, info = self.obs.simulate(deconexion_action)
        self.obs._obs_env.update_grid(self.env)
        simul_obs, simul_r1, d1, info1 = self.obs.simulate(act1)
        simul_ob2, simul_r2, d2, info2 = self.obs.simulate(act2)
```
Indicating that `obs.simulate` doesn't reset lines status correctly on subsequent calls. It also seems to happen only on this particular dataset (At least it doesn't on rte_case14_realistic)

**Comment by BDonnot** [comment]
This issue has been fixed in the development version. We will push the modifications in the new version as soon as possible.

**Comment by BDonnot** [comment]
Issue addressed and fixed in 1.0.0
